Runtime/Triangulator.cs

```diff
@@ -283,49 +288,485 @@ public void Dispose()
+            public bool IsAcuteAngle(Coord a, Coord b, Coord c);
+            public LengthSq Area2(int i, int j, int k, ReadOnlySpan<Coord> positions);
+            public Circle<Coord, LengthSq> CalculateCircumCircle(int i, int j, int k, NativeArray<Coord> positions);
+            public LengthSq CircumRadiusSq(Coord a, Coord b, Coord c);
+            public Coord CircumCenter(Coord a, Coord b, Coord c);
+            public int Orient2dFast(Coord a, Coord b, Coord c);
+            public bool InCircle(Coord a, Coord b, Coord c, Coord p);
+            public bool PointInsideTriangle(Coord p, Coord a, Coord b, Coord c);
+            public bool PointLineSegmentIntersection(Coord a, Coord b0, Coord b1);
+            public bool EdgeEdgeIntersection(Coord a0, Coord a1, Coord b0, Coord b1);
+            public bool IsConvexQuadrilateral(Coord a, Coord b, Coord c, Coord d);
+
+            public bool IsValidCoordinate(Coord coord);
+            public bool Equals (Coord a, Coord b);
+            public float Distance (Coord a, Coord b);
+            public LengthSq DistanceSq(Coord a, Coord b);
+            public bool LengthSmaller (LengthSq lhs, LengthSq rhs);
+            public LengthSq MultiplyLengthSq (LengthSq lhs, float multiplier);
+
+            public AffineTransform2D CalculatePCATransformation(NativeArray<Coord> positions);
+            public AffineTransform2D CalculateLocalTransformation(NativeArray<Coord> positions);
+
+            public void Transform (AffineTransform2D transform, NativeArray<Coord> points);
+            public void Transform (AffineTransform2D transform, [NoAlias] NativeArray<Coord> points, [NoAlias] NativeArray<Coord> outPoints);
+            public void InverseTransform (AffineTransform2D transform, [NoAlias] NativeArray<Coord> points, [NoAlias] NativeArray<Coord> outPoints);
+            public LengthSq InfLengthSq { get; }
+
+            public int HashKey(Coord p, Coord origin, int hashSize);
+            public void BoundingBox (ReadOnlySpan<Coord> positions, out Coord min, out Coord max, out Coord center);
+            public Coord Lerp(Coord a, Coord b, float t);
+            public bool SmallestInnerAngleIsBelowThreshold(Coord a, Coord b, Coord c, float angleThreshold);
+        }
+
         #region Jobs
+
+        public struct InputData<Coord> where Coord : unmanaged
+        {
+            public NativeArray<Coord> Positions;
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<int> ConstraintEdges;
+            [NativeDisableContainerSafetyRestriction]
+            public NativeArray<Coord> HoleSeeds;
+        }
+
+        public struct OutputData<Coord> where Coord : unmanaged {
+            [NativeDisableContainerSafetyRestriction]
+            public NativeList<Coord> Positions;
+            public NativeList<int> Triangles;
+            public NativeReference<Status> Status;
+        }
```

Comment:
Please make these structs `private`/`internal`. We do not want to include them in API.

Reply:
If one should be able to use the api from burst, struct inputs/outputs are necessary.

Reply:
As disscussed in the previous thread: `Triangulator` is not designed to call it from Burst. The future `NativeTriangulator` is planned to have this feature.